src/components/ViolationMessages.tsx

```diff
@@ -0,0 +1,23 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import ViolationsUtils from '@libs/ViolationsUtils';
+import useThemeStyles from '@styles/useThemeStyles';
+import {TransactionViolation} from '@src/types/onyx';
+import Text from './Text';
+
+export default function ViolationMessages({violations}: {violations: TransactionViolation[]}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const violationMessages = useMemo(() => violations.map((violation) => [violation.name, ViolationsUtils.getViolationTranslation(violation, translate)]), [translate, violations]);
+
+    return (
```

Comment:
NAB, but should we return `''` when there are no violationMessages, so that we don't render empty fragments `<> </>`?
```suggestion
    return violations.length === 0 ? '' : (
```

Reply:
I would say no -- a fragment isn't actually a component, so an empty fragment won't create a DOM node or anything. It'll just disappear.

I'm not even sure the fragment is necessary. We can just return the array of nodes.

Reply:
```tsx
return violationMessages.map(([name, message]) => (
        <View key={`violationMessages.${name}`}>
            <Text style={[styles.ph5, styles.textLabelError]}>{message}</Text>
        </View>
    ));
    ```